flow/beta/cli/project.go

```diff
@@ -113,145 +132,52 @@ func generateEmulatorServiceAccount() (*Account, *keys.HexAccountKey) {
 
 	serviceAccountKey := keys.NewHexAccountKeyFromPrivateKey(0, crypto.SHA3_256, privateKey)
 
-	return &Account{
-		name:    defaultEmulatorServiceAccountName,
-		address: flow.ServiceAddress(flow.Emulator),
-		chainID: flow.Emulator,
-		keys: []keys.AccountKey{
-			serviceAccountKey,
-		},
-	}, serviceAccountKey
-}
-
-func newProject(conf *config.Config) (*Project, error) {
-	accounts, err := accountsFromConfig(conf)
-	if err != nil {
-		return nil, err
+	return &config.Account{
+		Name:    defaultEmulatorServiceAccountName,
+		Address: flow.ServiceAddress(flow.Emulator),
+		ChainID: flow.Emulator,
+		Keys: []config.AccountKey{{
+			Type:     serviceAccountKey.Type(),
+			Index:    serviceAccountKey.Index(),
+			SigAlgo:  serviceAccountKey.SigAlgo(),
+			HashAlgo: serviceAccountKey.HashAlgo(),
+			Context:  serviceAccountKey.ToConfig().Context,
+		}},
 	}
-
-	return &Project{
-		conf:     conf,
-		accounts: accounts,
-	}, nil
-}
-
-func (p *Project) Host(network string) string {
-	return p.conf.Networks[network].Host
 }
 
-func (p *Project) EmulatorConfig(profile string) config.EmulatorConfigProfile {
-	return p.conf.Emulator[profile]
-}
-
-func (p *Project) AccountByAddress(address flow.Address) *Account {
-	for _, account := range p.accounts {
-		if account.Address() == address {
-			return account
-		}
-	}
-
-	return nil
-}
+func (p *Project) GetContractsByNetwork(network string) []Contract {
+	contracts := make([]Contract, 0)
 
-func (p *Project) Save() {
-	p.conf.Accounts = accountsToConfig(p.accounts)
+	// get deploys for specific network
+	for _, deploy := range p.Config.Deploy.GetByNetwork(network) {
+		account := p.Config.Accounts.GetByName(deploy.Account)
+		// go through each contract for this deploy
+		for _, contractName := range deploy.Contracts {
+			c := p.Config.Contracts.GetByName(contractName)
 
-	err := config.Save(p.conf, DefaultConfigPath)
-	if err != nil {
-		Exitf(1, "Failed to save project configuration to \"%s\"", DefaultConfigPath)
-	}
-}
+			contract := Contract{
+				Name:   c.Name,
+				Source: path.Clean(c.Source), //TODO: not necessary path - future improvements will include urls... REF: move this to config as validation and parsing
+				Target: account.Address,
+			}
 
-func (p *Project) getTargetAddress(target string) flow.Address {
-	for _, account := range p.accounts {
-		if account.name == target {
-			return account.address
+			contracts = append(contracts, contract)
 		}
 	}
 
-	return flow.HexToAddress(target[2:])
-}
-
-type Account struct {
-	name    string
-	address flow.Address
-	chainID flow.ChainID
-	keys    []keys.AccountKey
```

Comment:
Have you given thought to how `keys.AccountKey` would fit into the new structure? Is it still useful to have keys defined in their own package?

I created the `keys` package to avoid polluting the config package with business logic, and also to isolate some of the more sensitive crypto code.

Reply:
Will look into this